part-spec/Registers.json

```diff
@@ -0,0 +1,88 @@
+{
+    "registerDefiniton": {
+        "type": "object",
+        "required": [
+            "registerName",
+            "registerAddressOffset",
+            "registerSize"
+        ],
+        "properties": {
+            "registerName": {
+                "description": "Name of a register",
+                "type": "string"
+            },
+            "registerLongName": {
+                "description": " Full Name of a register",
+                "type": "string"
+            },
+            "registerAddressOffset": {
+                "description": "Address of a register",
+                "type": "number",
+                "comment": "Hex numeral system"
+            },
+            "registerSize": {
+                "description": "register Size",
+                "type": "number",
+                "comment": "units of bytes"
+            },
+            "registerType": {
+                "description": "Type of a register",
+                "type": "string",
+                "comment": "examples include but are not limited to data registers, address registers"
+            },
+            "registerResetValue": {
+                "description": "Reset value of a register",
+                "type": "number",
+                "comment": "Hex numeral system"
+            },
+            "registerValue": {
+                "description": "Value of a register",
+                "type": "number",
+                "comment": "Hex numeral system"
+            },
+            "registerAccessType": {
+                "description": " Access type ( Read, Write) of a Register",
+                "type": "string",
+                "comment": " examples include but are not limited to Read Only, WRite, Read Write"
```

Comment:
If there are limited possible values, can we make the `type` something that is something like an enum type instead of a generic string type?

(honest question, I am not sure if you can do enum-like-things with this JSON schema language.)

Reply:
Yes, JSON supports enum type. The concern I have is that i don't have an exhaustive list of the register types so I'd rather not exclude some valid values.

Reply:
Make it Enum [ R, RW]